src/ot/kern.rs

```diff
@@ -8,6 +8,18 @@ use super::TableIndex;
 use super::apply::ApplyContext;
 use super::matching::SkippyIter;
 
+pub fn has_kerning(face: &Face) -> bool {
+    unsafe { ffi::rb_ot_layout_has_kerning(face.as_ptr()) != 0 }
+}
+
+pub fn has_machine_kerning(face: &Face) -> bool {
+    unsafe { ffi::rb_ot_layout_has_machine_kerning(face.as_ptr()) != 0 }
+}
+
+pub fn has_cross_kerning(face: &Face) -> bool {
+    unsafe { ffi::rb_ot_layout_has_cross_kerning(face.as_ptr()) != 0 }
+}
+
 pub fn kern(plan: &ShapePlan, face: &Face, buffer: &mut Buffer) {
     unsafe { ffi::rb_ot_layout_kern(plan.as_ptr(), face.as_ptr(), buffer.as_ptr()); }
 }
```

Comment:
These four methods are still [implemented in C++](https://github.com/RazrFalcon/rustybuzz/blob/master/harfbuzz/src/hb-kern.cc) because they depend on the `kern` table (which in turn depends on `kerx`) and the `rb_aat_apply_context_t`.

Reply:
What is the question?

Reply:
No question. That was an answer to your question whether we still have kerning in C++.

Reply:
Oh, I see. So we cannot have a pure Rust version even with AAT feature-gated?

Reply:
Only if we put kerning behind the `aat` feature _or_ put it behind a `kern` feature that depends on the `aat` feature _or_ port it over from ttf-parser.

Reply:
Hmm... In this case I suggest using your own fork for now, and I will try to port AAT as soon as possible.

Reply:
Okay, then I'll do that. I will also have a look at making GSUB/GPOS faster.

Is there anything still blocking this PR from getting merged?

Reply:
Nope. Thanks. I will start working on AAT next week. Just to make sure we would not overlap.